wgengine/router/router_linux.go

```diff
@@ -430,6 +433,44 @@ func (r *linuxRouter) Set(cfg *Config) error {
 	return multierr.New(errs...)
 }
 
+var warnStatefulFilteringWithDocker = health.NewWarnable()
+
+func (r *linuxRouter) updateStatefulFilteringWithDockerWarning(cfg *Config) {
+	// If stateful filtering is disabled, clear the warning.
+	if !r.statefulFiltering {
+		r.health.SetWarnable(warnStatefulFilteringWithDocker, nil)
+		return
+	}
+
+	// If this node isn't a subnet router or exit node, then we would never
```

Comment:
I _think_ this isn't right e.g. in the case the container is being used as an egress proxy where it does a SNAT. This is what we are working around in the k8s operator too.